src/components/views/elements/Slider.tsx

```diff
@@ -0,0 +1,135 @@
+
+type IProps = {
+        // A callback for the selected value
+        onSelectionChange: (value: number) => void;
+
+        // The current value of the slider
+        value: number;
+
+        // The range and values of the slider
+        // Currently only supports an ascending, constant interval range
+        values: number[];
+
+        // A function for formatting the the values
+        displayFunc: (value: number) => string;
+
+        // Whether the slider is disabled
+        disabled: boolean;
+}
+
+export default class Slider extends React.Component<IProps> {
+    // offset is a terrible inverse approximation.
+    // if the values represents some function f(x) = y where x is the
+    // index of the array and y = values[x] then offset(f, y) = x
+    // s.t f(x) = y.
+    // it assumes a monotonic function and interpolates linearly between
+    // y values.
+    // Offset is used for finding the location of a value on a
+    // non linear slider.
+    _offset(values: number[], value: number): number {
+        // the index of the first number greater than value.
+        let closest = values.reduce((prev, curr) => {
+            return (value > curr ? prev + 1 : prev);
+          }, 0);
+
+        // Off the left
+        if (closest === 0) {
+            return 0;
+        }
+
+        // Off the right
+        if (closest === values.length) {
+            return 100;
+        }
+
+        // Now
+        const closestLessValue = values[closest - 1];
+        const closestGreaterValue = values[closest];
+
+        const intervalWidth = 1 / (values.length - 1);
+
+        const linearInterpolation = (value - closestLessValue) / (closestGreaterValue - closestLessValue)
+
+        return 100 * (closest - 1 + linearInterpolation) * intervalWidth
+
+    }
+
+    render(): React.ReactNode {
+        const dots = this.props.values.map(v =>
+            <Dot active={v <= this.props.value}
+                 label={this.props.displayFunc(v)}
+                 onClick={this.props.disabled ? () => {} : () => this.props.onSelectionChange(v)}
+                 key={v}
+                 disabled={this.props.disabled}
+            />);
+
+        const offset = this._offset(this.props.values, this.props.value);
+
+        return <div className="mx_Slider">
+            <div>
+                <div className="mx_Slider_bar">
+                    <hr />
+                    { this.props.disabled ?
+                        null :
+                        <div className="mx_Slider_selection">
+                            <div className="mx_Slider_selectionDot" style={{left: "calc(-0.55em + " + offset + "%)"}} />
+                            <hr style={{width: offset + "%"}} />
+                        </div>
+                    }
+                </div>
+                <div className="mx_Slider_dotContainer">
+                    {dots}
+                </div>
+            </div>
+        </div>;
+    }
+}
+
+type DotIProps = {
```

Comment:
(Interfaces start with `I` historically)
```suggestion
interface IDotsProps {
```